File: src/main/java/com/github/sheauoian/croissantsys/common/user/StatusType.java
package com.github.sheauoian.croissantsys.common.user;

public enum StatusType {
    STRENGTH("攻撃力",3.8d),
    DEFENCE("防御力", 4.5d),
    MAX_HP("最大体力", 3.5d);

    private final String displayName;
    private final double subVolume;

    StatusType(String displayName, double subVolume) {
        this.displayName = displayName;
        this.subVolume = subVolume;
    }

    public double subVolume() {
        return this.subVolume;
    }

    public String displayName() {
        return this.displayName;
    }
}

File: src/main/java/com/github/sheauoian/croissantsys/util/status/StatusType.kt
package com.github.sheauoian.croissantsys.util.status

enum class StatusType(val displayName: String, val baseVolume: Double) {
    STR("攻撃力", 1.0),
    DEF("防御力", 1.0),
    MAX_HP("体力", 2.0),
    MAGIC_EFFECT("スキル性能", 1.0),
    CRITICAL_RATE("クリティカル確率", 0.0);
    companion object {
        fun valueOfOrNull(value: String?): StatusType? {
            return try {
                valueOf(value!!)
            } catch (e: IllegalArgumentException) {
                null
            }
        }

        fun getRandom(exclude: List<StatusType>): StatusType {
            return entries.filterNot { it in exclude } .random()
        }
    }
}